### DISCUSSION TITLE
Did anyone try to extract data from the FusionSolar Website? (instead of Kiosk or OpenAPI)

### DISCUSSION THREAD

**Discussion by SpraxDev** [tijsverkoyen/HomeAssistant-FusionSolar#164] [General]
Hi, we just recently got our solar system and access to the FusionSolar app/website.

Because of the (stupid) API restrictions requiring the installer to create an account but unable to create enough accounts for all of their customers, I was thinking if I would be able to extract the data from website directly as the Kiosk URL is sadly missing the battery.

I was wondering why I don't find anything in that regard and if Huawei/FusionSolar blocked accounts or something because of that.
If they limit their API this hard, I could imagine them not being happy about such kind usage?

The Website essentially just uses JSON-API-Endpoints itself - So I'd just login, get the cookie(s) for auth, maybe extract some csrf tokens or whatever from the HTML if needed, and use the `station-detail` endpoint as it seems to include all/most of the information I am interested in that the Kiosk is lacking.

---

Any ideas/infos/warnings in that regard? :o

**Comment by tijsverkoyen** [comment]
Maybe https://github.com/tijsverkoyen/HomeAssistant-FusionSolar/issues/162 is related?

**Reply by SpraxDev** [reply]
Oh yeah it is - Didn't see that 😅 
Maybe I checked the issues a couple days before I created this post and didn't check again.

I'm watching the issue now, thanks - I have no idea of Python and won't be contributing with that but I''ll certainty take a look at the endpoints and do some testing
